src/cuda-kernels/identities.cu

```diff
@@ -0,0 +1,43 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/master/LICENSE
+
+#include "awkward/cuda-kernels/cuda_identities.h"
+#include <stdio.h>
+
+__global__
+void cuda_Identities32_toIdentities64(
+  int64_t* toptr,
+  const int32_t* fromptr,
+  int64_t length) {
+  int64_t block_id = blockIdx.x + blockIdx.y * gridDim.x + gridDim.x * gridDim.y * blockIdx.z;
+  int64_t thread_id = block_id * blockDim.x + threadIdx.x;
+  if(thread_id < length) {
+    toptr[thread_id] = (int64_t)(fromptr[thread_id]);
+  }
+}
+
+ERROR awkward_cuda_Identities32_to_Identities64(
+  int64_t* toptr,
+  const int32_t* fromptr,
+  int64_t length,
+  int64_t width) {
+
+  dim3 blocks_per_grid;
+  dim3 threads_per_block;
+
+  if (length > 1024) {
+    blocks_per_grid = dim3(ceil((length) / 1024.0), 1, 1);
+    threads_per_block = dim3(1024, 1, 1);
+  } else {
+    blocks_per_grid = dim3(1, 1, 1);
+    threads_per_block = dim3(length, 1, 1);
+  }
+
+  cuda_Identities32_toIdentities64<<<blocks_per_grid, threads_per_block>>>(
+    toptr,
+    fromptr,
+    length * width);
```

Comment:
I would prefer to pass `length` and `width` as separate arguments since it is easier to reason about in the code generation.

Reply:
Okay, I did this to reduce computation within the kernel, since global memory is slow. Doing this will not break things, but if `nvcc` doesn't optimise this there could be a slow down. But we can catch all these cases after the generation, right?

Reply:
When you say, "did this," what "this" are you referring to? What are you expecting to be slow and how did you try to fix it (or not)?

In general, we can put off performance tweaks for later, though we should design around a structure that can be performant.

Reply:
So, by "did this" I was referring to how I passed `length * width` as one parameter to do the multiplication on the host side, rather than on the device side(multiple times for each thread).

Reply:
Yes, we should absolutely not worry about whether a multiplication is done on the host side or device side, and whether it's done once per thread or once for all. Addition and multiplication are the two fastest operations a processor can do and they're usually vastly overshadowed by time spent fetching memory. On a GPU, the memory-fetching is less significant because global memory is essentially like a CPU's cache—it's close to the processor—but still, it's unlikely that the other work these kernels need to do (especially the parallel prefix sum ones) will be close to the speed of multiplication.

Also, it's not "structural," we can easily change it in the future, in response to profiling tests. So it's not something that we'll worry about right now.